utils/network.go

```diff
@@ -6,151 +6,229 @@ import (
-		*n = GOERLI
-	case "GOERLI2", "goerli2":
-		*n = GOERLI2
-	case "INTEGRATION", "integration":
-		*n = INTEGRATION
-	default:
+	predefinedNetworks := map[string]Network{
+		"MAINNET":     MAINNET,
+		"mainnet":     MAINNET,
+		"GOERLI":      GOERLI,
+		"goerli":      GOERLI,
+		"GOERLI2":     GOERLI2,
+		"goerli2":     GOERLI2,
+		"INTEGRATION": INTEGRATION,
+		"integration": INTEGRATION,
+	}
+
+	if network, ok := predefinedNetworks[s]; ok {
+		*n = network
+		return nil
+	}
+
+	return n.setCustomNetwork(s)
+}
+
+func (n *Network) setCustomNetwork(s string) error {
+	*n = Network{}
+	elems := strings.Split(s, ",")
+
+	if !(elems[0] == "custom" || elems[0] == "CUSTOM") {
 		return ErrUnknownNetwork
 	}
+
+	if len(elems) != 9 { /* number of required fields in Network struct */ //nolint:gomnd
+		return ErrNetworkParamsNotSet
+	}
+
+	n.name = elems[0]
+	n.baseURL = elems[1]
+	n.chainID = elems[2]
+
+	if elems[3] != "" {
+		l1ChainID, success := new(big.Int).SetString(elems[3], 10) //nolint:gomnd
+		if !success {
+			return errors.New("L1 Chain ID must be an integer (base 10)")
+		}
+		n.l1ChainID = l1ChainID
+	}
+
+	if elems[4] != "" {
+		n.coreContractAddress = common.HexToAddress(elems[4])
+	}
+	requiredFields := []struct {
+		value string
+		err   error
+	}{
+		{elems[5], ErrNetworkNoFallbackAddr},
+		{elems[6], ErrNetworkNoFirst07Block},
+		{elems[7], ErrNetworkNoUnverifRange},
+		{elems[8], ErrNetworkNoUnverifRange},
+	}
+
+	for _, field := range requiredFields {
+		if field.value == "" {
+			return field.err
+		}
+	}
+
+	n.blockHashMetaInfo = &blockHashMetaInfo{}
+
+	var err error
+	if n.blockHashMetaInfo.FallBackSequencerAddress, err = new(felt.Felt).SetString(elems[5]); err != nil {
+		return ErrNetworkSetFallbackAddr
+	}
+	if n.blockHashMetaInfo.First07Block, err = strconv.ParseUint(elems[6], 10, 64); err != nil {
+		return ErrNetworkSetFirst07Block
+	}
+	var start, end uint64
+	if start, err = strconv.ParseUint(elems[7], 10, 64); err != nil {
+		return ErrNetworkSetUnverifRangeStart
+	}
+	if end, err = strconv.ParseUint(elems[8], 10, 64); err != nil {
+		return ErrNetworkSetUnverifRangeEnd
+	}
+	n.blockHashMetaInfo.UnverifiableRange = []uint64{start, end}
 	return nil
 }
 
-func (n *Network) Type() string {
+func (n Network) Type() string {
 	return "Network"
 }
 
 func (n *Network) UnmarshalText(text []byte) error {
 	return n.Set(string(text))
 }
 
-// baseURL returns the base URL without endpoint
-func (n Network) baseURL() string {
-	switch n {
-	case GOERLI:
-		return "https://alpha4.starknet.io/"
-	case MAINNET:
-		return "https://alpha-mainnet.starknet.io/"
-	case GOERLI2:
-		return "https://alpha4-2.starknet.io/"
-	case INTEGRATION:
-		return "https://external.integration.starknet.io/"
-	default:
-		// Should not happen.
-		panic(ErrUnknownNetwork)
-	}
-}
-
 // FeederURL returns URL for read commands
 func (n Network) FeederURL() string {
-	return n.baseURL() + "feeder_gateway/"
+	return n.baseURL + "feeder_gateway/"
```

Comment:
Do you think we should have a `feederURL` field instead of `baseURL`? As it stands, custom networks must include sequencer endpoints on the `feeder_gateway` path. Same for `gatewayURL`.

Reply:
I replaced the baseURL with FeederURL and GatewayURL fields.